### PR TITLE
add example generation

### PR DESCRIPTION
**Pull request by wistefan**
In order to reduce boilerplate code in api tests, the generator can now produce Example Instances of all model objects. The values will be set as following:
1. use the example from the spec
2. use the default from the spec
3. use a type specific default
4. is nothing applies, set value to null

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +56,7 @@ Include in your `pom.xml`:
 * openApiNullable: add container `JsonNullable` to model objects that are nullable (default `true`)
 * supportAsync: use reactivex return types, see [Reactive HTTP Request Processing](https://docs.micronaut.io/latest/guide/index.html#reactiveServer)
 * pagable: generate provider for pagable (default `false`)
+* generateExamples: generate example objects for the model, based on example and default values
```
Comment (sschnabe) [inline README.md:59]: document default value

Comment (sschnabe) [inline README.md:59]:
```suggestion
* generateExamples: generate example objects for the model, based on example and default values (default `true`)
```

File: gen/test/java/testmodel/examples/ModelTestExample.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package testmodel.examples;
+
+import testmodel.examples.Model.*;
```
Comment (sschnabe) [inline gen/test/java/testmodel/examples/ModelTestExample.java:3]: is this Import necessary?

File: src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java [hunk 3]
```diff
@@ -153,12 +162,13 @@ public String getName() {
 	}
 
 	@Override
-	public void postProcess() {}
+	public void postProcess() {
```
Comment (sschnabe) [inline src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java:165]: why reformatting? (there are other reformats in this file)

**Review by sschnabe — CHANGES_REQUESTED** [review]

File: gen/test/java/testmodel/examples/ModelTestExample.java (hunk 2 above)
Comment (wistefan) [reply gen/test/java/testmodel/examples/ModelTestExample.java:3]: Not everywhere, but for cases that use Embedded Enums(see for example: https://github.com/kokuwaio/micronaut-openapi-codegen/blob/328fe36f157e0b6c7807df4fc658c0eb122ccc85/gen/test/java/testmodel/examples/EnumerationModelTestExample.java). If you have an idea on how to identifiy such classes in the mustache we can put it only on those, but I didn't find one.

File: src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java (hunk 3 above)
Comment (wistefan) [reply src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java:165]: Same as always, conflicting code formatters;) I will fix.

File: README.md (hunk 1 above)
Comment (wistefan) [reply README.md]: done

File: src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java (hunk 3 above)
Comment (wistefan) [reply src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java]: done

**Comment by wistefan** [comment]
Do not merge. There is an issue with inheritance that I'm currently solving.

**Review by sschnabe — APPROVED** [review]

**Pull request merged**